### PR TITLE
Plugin Size Reduction

### PR DESCRIPTION
**Pull request by whatever4711**
Simply changing the base image for the plugin from Debian to Alpine reduces the size from 60MB to 10MB for the plugin. This is beneficial for all small devices like the RPi. Therefore, the `Dockerfile` is changed and the goal `gobuild_docker_%` is updated to use `golang:alpine3.14`.

### CODE REVIEW

**Comment by Skazza94** [comment]
Hi @whatever4711,
thanks for the PR. Seems good to have a smaller Network Plugin image. We used Debian while developing it for faster debugging, but now it is ok to switch to Alpine. @lorenzo93 will take care of it.

File: img-src/Dockerfile [hunk 1]
```diff
@@ -1,16 +1,10 @@
-FROM debian:buster
+FROM alpine
 
-RUN apt update
-RUN apt upgrade -y
-RUN DEBIAN_FRONTEND=noninteractive apt install -y \
-    bash \
+RUN apk add -U \
     iptables
```
Comment (lorenzo93) [inline img-src/Dockerfile]: To make the plugin work, we need both iptables and nftables.
Moreover, due to the switch [here](https://github.com/KatharaFramework/NetworkPlugin/blob/master/go-src/src/common_utils.go#L47) and [here](https://github.com/KatharaFramework/NetworkPlugin/blob/c0bc61a5ad16de974c963e4f33e559e1dc51232a/go-src/src/common_utils.go#L51) we need also `dpkg` installed into the container (`update-alternatives` is a part of the `dpkg` package)
So this should install all this requirements and be something like this
`apk add -U iptables dpkg nftables`

File: img-src/Dockerfile [hunk 2]
```diff
@@ -1,16 +1,10 @@
...
-RUN apt clean
```
Comment (lorenzo93) [inline img-src/Dockerfile:8]: You should add the equivalent for `apk` to have the image smaller.
According to [this thread](https://stackoverflow.com/questions/49118579/alpine-dockerfile-advantages-of-no-cache-vs-rm-var-cache-apk) the best way is to run `apk add` with the `--no-cache` parameter

File: img-src/Dockerfile [hunk 3]
```diff
@@ -1,16 +1,10 @@
...
 
 WORKDIR /app
 
 COPY entrypoint.sh entrypoint.sh
-RUN chmod +x entrypoint.sh
-
```
Comment (lorenzo93) [inline img-src/Dockerfile:14]: To make `update-alternatives` working, the alternatives must be registered.
This are simple examples that I already tested in an alpine container:
- `update-alternatives --install /sbin/iptables iptables /sbin/iptables-legacy 1`
- `update-alternatives --install /sbin/iptables iptables /sbin/iptables-nft 2`

**Review by lorenzo93 — CHANGES_REQUESTED** [review]
Hi @whatever4711 ,

Your idea is really smart and I like it :)

The network plugin automatically switches at runtime between `iptables` and `nftables`. This is needed due to the compatibility with different linux distros (different distros uses different versions and they are not cross-compatible).

I've added to the review the changes to be compatible with all the distros.

File: img-src/Dockerfile (hunk 3 above)
Comment (lorenzo93) [reply img-src/Dockerfile:14]: I don't know how to comment on a file without selecting a line, so I commented a random line about this. ;)

**Comment by whatever4711** [comment]
@lorenzo93: I updated your suggestions on the latest commit. Thanks for your efforts.

**Review by lorenzo93 — APPROVED** [review]

**Comment by lorenzo93** [comment]
For me it is everything ok.

I'll merge and close this PR when I'll publish the new plugin with alpine (during next week).

Many thanks for the PR and changes.

**Comment by Skazza94** [comment]
Hi @whatever4711,
I pushed the new plugin containers which use Alpine.

Thanks for the PR.

Mariano.

**Pull request merged**